filters/tee/tee_test.go

```diff
@@ -0,0 +1,129 @@
+package tee
+
+import (
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/filters/filtertest"
+	"github.com/zalando/skipper/proxy/proxytest"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+var (
+	testTeeSpec        = NewTee()
+	teeArgsAsBackend   = []interface{}{"https://api.example.com"}
+	teeArgsWithModPath = []interface{}{"https://api.example.com", ".*", "/v1/"}
+)
+
+func TestTeeHostHeaderChanges(t *testing.T) {
+	f, _ := testTeeSpec.CreateFilter(teeArgsAsBackend)
+	fc := buildfilterContext()
+
+	rep, _ := f.(*tee)
+	modifiedRequest := cloneRequest(rep, fc.Request())
+	if modifiedRequest.Host != "api.example.com" {
+		t.Error("Tee Request Host not modified")
+	}
+
+	originalRequest := fc.Request()
+	if originalRequest.Host == "api.example.com" {
+		t.Error("Incoming Request Host was modified")
+	}
+}
+
+func TestTeeSchemeChanges(t *testing.T) {
+	f, _ := testTeeSpec.CreateFilter(teeArgsAsBackend)
+	fc := buildfilterContext()
+
+	rep, _ := f.(*tee)
+	modifiedRequest := cloneRequest(rep, fc.Request())
+	if modifiedRequest.URL.Scheme != "https" {
+		t.Error("Tee Request Scheme not modified")
+	}
+
+	originalRequest := fc.Request()
+	if originalRequest.URL.Scheme == "https" {
+		t.Error("Incoming Request Scheme was modified")
+	}
+}
+
+func TestTeeUrlHostChanges(t *testing.T) {
+	f, _ := testTeeSpec.CreateFilter(teeArgsAsBackend)
+	fc := buildfilterContext()
+
+	rep, _ := f.(*tee)
+	modifiedRequest := cloneRequest(rep, fc.Request())
+	if modifiedRequest.URL.Host != "api.example.com" {
+		t.Error("Tee Request Url Host not modified")
+	}
+
+	originalRequest := fc.Request()
+	if originalRequest.URL.Host == "api.example.com" {
+		t.Error("Incoming Request Url Host was modified")
+	}
+}
+
+func TestTeeWithPathChanges(t *testing.T) {
+	f, _ := testTeeSpec.CreateFilter(teeArgsWithModPath)
+	fc := buildfilterContext()
+
+	rep, _ := f.(*tee)
+	modifiedRequest := cloneRequest(rep, fc.Request())
+	if modifiedRequest.URL.Path != "/v1/" {
+		t.Errorf("Tee Request Path not modified, %v", modifiedRequest.URL.Path)
+	}
+
+	originalRequest := fc.Request()
+	if originalRequest.URL.Path != "/api/v3" {
+		t.Errorf("Incoming Request Scheme modified, %v", originalRequest.URL.Path)
+	}
+}
+
+type MyHandler struct {
+	name string
+	body string
+}
+
+func (h *MyHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	b, _ := ioutil.ReadAll(r.Body)
+	str := string(b)
+	h.body = str
+}
+
+func TestTeeEndToEndBody(t *testing.T) {
+	shadowHandler := &MyHandler{name: "shadow"}
+	shadowServer := httptest.NewServer(shadowHandler)
+	shadowUrl := shadowServer.URL
+	defer shadowServer.Close()
+
+	originalHandler := &MyHandler{name: "original"}
+	originalServer := httptest.NewServer(originalHandler)
+	originalUrl := originalServer.URL
+	defer originalServer.Close()
+
+	routeStr := "route1: * -> Tee(\"" + shadowUrl + "\")" + " -> " + "\"" + originalUrl + "\";"
```

Comment:
consider using the backtick "`" strings for the case of the escaped doublequotes '"'. also, this can be the right use case for fmt.Sprintf() https://godoc.org/fmt#Sprintf